### ISSUE TITLE
Conversion of RTF/DOCX/ODT is not proper for some files

### ISSUE THREAD

**Issue by skrchnavy** [opencats/OpenCATS#150]
This is follow up of PR #149 / bug #105.

Implementation of conversion works OK when the file is simple and 'standard'
For some nonstandard cases it could cause strange behavior.
These 3 file types are processed different way comparing to DOC or PDF.

Example - when pdf file is renamed to rdf (extension changed) and then attahced as rdf to openCATS, it can cause openCATS hangup.

**Comment by RussH** [comment]
why is a PDF renamed to rdf? surely then it wouldn't open in a PDF viewer? I'd have thought that filtering wouldn't touch anything that isn't pdf / doc / docx / odf, etc, etc

**Comment by skrchnavy** [comment]
I accidentally wrote rdf, shall be rtf.

This is case when user wants to break functionality, (s)he can upload document with not proper extension, can cause described result.

**Comment by RussH** [comment]
ahh, so it's trying to use unRTF on a PDF and hangs (?)



----
Russ

**Comment by skrchnavy** [comment]
yes, something like that, don't remember now how i changed suffix of PDF file. We fixed just warnings in mentioned PR#149 and observed this issue.

**Comment by mlespiau** [comment]
This problem can be fixed by detecting the file type based on [it's magic numbers](https://en.wikipedia.org/wiki/List_of_file_signatures). PHP [file info function](http://php.net/manual/es/function.finfo-file.php) uses libmagic to do this.
